scripts/generate-flux-gpg-key.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+set +e
+
+if ! command -v aws &> /dev/null
+then
+  echo "Please install the AWS CLI."
+  exit 1
+fi
+
+if ! command -v pwgen &> /dev/null
+then
+  echo "Please install pwgen."
+  exit 2
+fi
+
+if ! command -v gpg &> /dev/null
+then
+  echo "Please install gpg."
+  exit 3
+fi
+
+# you'll also need the AWS CLI to be configured with credentials for our AWS account.
+
+AWS_DEFAULT_REGION=us-west-2
+FLUX_REALNAME="Flux SWF Client"
+FLUX_EMAIL="aws-flux-swf-client-owners@amazon.com"
+
+TEMPDIR=$(mktemp -d)
+echo "Working in ${TEMPDIR}..."
+
+echo "Generating passphrase..."
+FLUX_GPG_PASSPHRASE=$(pwgen -s 16 1)
+
+echo "Generating key..."
+gpg --homedir $TEMPDIR --batch --gen-key <<FOO
+    Key-Type: RSA
+    Key-Length: 4096
+    Name-Real: $FLUX_REALNAME
+    Name-Email: $FLUX_EMAIL
+    Expire-Date: 1y
+    Passphrase: $FLUX_GPG_PASSPHRASE
+    %commit
+    %echo done
+FOO
+
+# the output of "gpg --list-keys --with-colons" looks something like this:
+
+# $ gpg --homedir . --list-public-keys --with-colons
+# tru::1:1648579633:0:3:1:5
+# pub:-:4096:1:7C5DC8336806E60E:1607727690:1680116664::-:::scESC::::::23::0:
+# fpr:::::::::A9C2E0DFFED2658D75A64D7A7C5DC8336806E60E:
+# uid:-::::1648580664::5F4E53DF54AD0F951384C0F6051B09F3E2BE8496::AWS Flux SWF Client <aws-flux-swf-client-owners@amazon.com>::::::::::0:
+# sub:-:4096:1:D8B3CEEC5C428011:1607727690:1680116665:::::e::::::23:
+# fpr:::::::::FB6E627D30C5B1F4C2177B1BD8B3CEEC5C428011:
+
+# we want the hex string from that line just before the uid.
+
+echo "Retrieving key id..."
+FLUX_GPG_KEY_ID=$(gpg --homedir . --list-public-keys --with-colons | grep -B 1 $FLUX_EMAIL | head -n 1 | cut -d ':' -f 10)
```

Comment:
Pretty sure that you for got to use `$TEMPDIR` here instead of `.`

Reply:
also, when i ran the same command i got nothing back when doing that cut. is it supposed to be `-f 10`? 

i can get it with awk pretty easily with `... | awk -F':' '/pub/{ print $5 }'`

Reply:
Nice catch, I'll fix the homedir thing.

We discussed the other comment and determined that older versions of gpg need the parameter `--fingerprint` to include the fingerprint, so I'll include it here just to be safe.

Reply:
Nice catch, I'll fix the homedir thing.

We discussed the other comment and determined that older versions of gpg need the parameter `--fingerprint` to include the fingerprint, so I'll include it here just to be safe.